src/main/java/com/google/api/codegen/go/GoGapicContext.java

```diff
@@ -494,19 +500,22 @@ private GoImport createMessageImport(MessageType messageType) {
    * indentation within the 'import' section in Go file.
    */
   public Iterable<String> getImports(Interface service) {
+    Set<GoImport> standard = getStandardImports(service);
     TreeSet<GoImport> thirdParty = new TreeSet<>();
 
     // Add non-service-specific imports.
     thirdParty.add(GoImport.create("golang.org/x/net/context"));
     thirdParty.add(GoImport.create("google.golang.org/grpc"));
-    thirdParty.add(GoImport.create("google.golang.org/grpc/codes"));
     thirdParty.add(GoImport.create("google.golang.org/grpc/metadata"));
     thirdParty.add(GoImport.create(GAX_PACKAGE_BASE, "gax"));
     thirdParty.add(GoImport.create("google.golang.org/api/option"));
     thirdParty.add(GoImport.create("google.golang.org/api/transport"));
 
+    if (standard.contains(GoImport.create("time"))) {
+      thirdParty.add(GoImport.create("google.golang.org/grpc/codes"));
```

Comment:
Wouldn't it be better to key this include directly and independently off of whether retry is being used, rather than assuming that whatever uses "time" also uses "codes"? If we ever require "time" for any other reason, it will bring in "codes" too, potentially unnecessarily.

Maybe simply check once for retry and set a bool, and check the bool in both import setting locations?